### PR TITLE
Add single emelemnt set support, fix maps compare

### PR DESCRIPTION
**Pull request by roytman**
Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

### CODE REVIEW

File: pkg/libovsdb/encoding_test.go [hunk 1]
```diff
@@ -2,48 +2,214 @@ package libovsdb
...
+	for _, elem := range set2.GoSet {
+		switch elem.(type) {
+		case UUID:
+			uuid := elem.(UUID)
+			res2[uuid.GoUUID] = true
+		default:
+			s := fmt.Sprintf("%v", elem)
+			res2[s] = true
+		}
+	}
```
Comment (hunchback) [inline pkg/libovsdb/encoding_test.go:151]: this is a total code duplication of the block above, please use functions ...

File: pkg/libovsdb/encoding_test.go [hunk 2]
```diff
@@ -2,48 +2,214 @@ package libovsdb
...
+		jsonExpectedOutput: `"aa"`,
+	},
+	{
+		objInput:           []string{`aa`, `bb`},
+		jsonExpectedOutput: `["set",["aa","bb"]]`,
+	},
+	{
+		objInput:           [2]string{`aa`, `bb`},
+		jsonExpectedOutput: `["set",["aa","bb"]]`,
+	},
```
Comment (hunchback) [inline pkg/libovsdb/encoding_test.go:62]: redundant, this is exactly as above (after go compilation its' identical).

File: pkg/libovsdb/encoding_test.go [hunk 3]
```diff
@@ -2,48 +2,214 @@ package libovsdb
...
+	{
+		objInput:           []int{10, 15},
+		jsonExpectedOutput: `["set",[10,15]]`,
+	},
+	{
+		objInput:           []uint16{10, 15},
+		jsonExpectedOutput: `["set",[10,15]]`,
+	},
```
Comment (hunchback) [inline pkg/libovsdb/encoding_test.go:70]: not sure why you are testing all simple types? if you test for string then no need to test for int, int32, float32, etc, etc. this adds no value as the focus here is to the OvsSet (which can contain anything) not serialization of all simple tests (something that is handled by standard json encoder/decoder).

**Comment by hunchback** [comment]
@roytman some higher level comments:

(1)
There is a lot of redundancy (for example no need to test for all types of set elements just use one simple type such as string).

(2)
There is code duplication - use helper functions even if this is test code

(3)
You separated Mashal and Unmarshal but combining both into a single test makes a lot of sense and enables to test both.

(4)
You need to have a standard (sorted) representation in json of OvsSet and ovsMap once this is the case all needed is to perform a string comparison. By doing this you eliminate most of the code complexity.

File: pkg/libovsdb/encoding_test.go (hunk 2 above)
Comment (roytman) [reply pkg/libovsdb/encoding_test.go:62]: No, they are not, one case is slice and another is array

File: pkg/libovsdb/encoding_test.go (hunk 1 above)
Comment (roytman) [reply pkg/libovsdb/encoding_test.go]: the function was removed

**Comment by roytman** [comment]
@hunchback I think I resolved all your comments, most of the set tests are to validate corner cases such as single element, array of elements, uuid, named-uuid ....
We don't need a standard (sorted) representation in json of OvsSet and ovsMap, there is `assert.ElementsMatch` instead.

File: pkg/libovsdb/encoding_test.go (hunk 2 above)
Comment (hunchback) [reply pkg/libovsdb/encoding_test.go:62]: ok
